tests/test_region_aggregation.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+from nomenclature.region_mapping_models import RegionAggregationMapping
+from jsonschema.exceptions import ValidationError
+
+from conftest import TEST_DATA_DIR
+
+test_folder = TEST_DATA_DIR / "region_aggregation"
+
+
+def test_mapping():
+    mapping_file = "working_mapping.yaml"
+    # Test that the file is read and represented correctly
+    ram = RegionAggregationMapping.create_from_region_mapping(
+        test_folder / mapping_file
+    )
+    reference = {
+        "model": "model_a",
+        "native_regions": [
+            {"name": "region_a", "rename": "alternative_name_a"},
+            {"name": "region_b", "rename": "alternative_name_b"},
+            {"name": "region_c", "rename": None},
+        ],
+        "common_regions": [
+            {
+                "name": "common_region_1",
+                "constituent_regions": [
+                    {"name": "region_a", "rename": None},
+                    {"name": "region_b", "rename": None},
+                ],
+            },
+            {
+                "name": "common_region_2",
+                "constituent_regions": [{"name": "region_c", "rename": None}],
+            },
+        ],
+    }
+    assert ram.dict() == reference
+
+
+@pytest.mark.parametrize(
+    "file, error_type, error_msg_pattern",
+    [
+        (
+            "illegal_mapping_invalid_format_dict.yaml",
+            ValidationError,
+            ".*common_region_1.*not.*'array'.*",
+        ),
+        (
+            "illegal_mapping_illegal_attribute.yaml",
+            ValidationError,
+            "Additional properties are not allowed.*",
+        ),
+        (
+            "illegal_mapping_conflict_regions.yaml",
+            ValueError,
+            ".*Conflict between \(renamed\).*common_region_1.*",
```

Comment:
Could work to just add `r` before the string, see https://www.flake8rules.com/rules/W605.html